### DISCUSSION TITLE
Possible to decrease update rate to 1 second?

### DISCUSSION THREAD

**Discussion by N1nja98** [jomjol/AI-on-the-edge-device#1979] [General] [answered]
Hello, I'm thinking of implementing this for reading a lighted 7-segment display that changes about every 8 seconds. The problem I see is that the default firmware only supports down to 1 minute update rates. Do you have any pointers on how I could optimize the firmware or neural network (perhaps by reducing the variety of recognized digit types) in order to reduce processing time to achieve 1 second update intervals? 

Below is an image of the 7 segment display I'm trying to read. Any help would be appreciated.
![IMG_4323](https://user-images.githubusercontent.com/47512532/216796878-478973ca-0d40-45c0-b8d5-59bd298063ad.jpg)

**Comment by jomjol** [comment, marked as answer]
In principal the ESP32 has some limits with going up in recognition rate.
There are some special tricks and optimizations in the CNN, the flow and the way you implement it, that allows you to go down to about 1s readout rate, which I offer in a commercial branch of this idea and do not support here.
If it is such a clear digit I would go another way: simply readout the intensity of the 7 segment from the pixels and judge back to the number. That will be very fast.

**Reply by N1nja98** [reply]
I appreciate the prompt and informative response, and also respect your choice in reserving it to commercial. 

I did observe and consider that the neural network approach might be a bit overkill for my use case...

Just to clarify, on your suggestion of reading the intensity of the pixels, as I understand the process would involve mapping the individual segments of the digits to pixels on the image and reading the intensity, thereby determining the number. Is that what you had in mind? 

If so, and if you don't mind me asking, is there any native support or implementation for this in your code, where it would be beneficial to branch off of your code, or would I be better off starting from scratch?

Thank you in advance.

**Comment by jomjol** [comment]
Yes, you got my idea.
But there is no support or natural way to implement in the software, as it would need a complete new module and also a user interface to configure it. That would be a lot of work for a very special case.

I would suggest to start from scratch.

**Reply by N1nja98** [reply]
Ok gotcha, I'll go that route and start from scratch, thanks for the assistance.
